schematic/schemas/data_model_graph.py

```diff
@@ -226,6 +229,64 @@ def get_adjacent_nodes_by_relationship(
 
         return list(nodes)
 
+    def get_component_node_validation_rules(
+        self, manifest_component: str, node_display_name: str
+    ) -> list[str]:
+        """Get valdation rules for a given node and component.
+        Args:
+            manifest_component, str: manifest component display name that the node belongs to.
+            node_display_name, str: node display name for the node being queried.
+        Returns:
+            validation_rules, list[str]: validation rules list for a given node and component.
+        """
+        # get any additional validation rules associated with this node (e.g. can this node be mapped to a list of other nodes)
+        node_validation_rules = self.get_node_validation_rules(
+            node_display_name=node_display_name
+        )
+
+        # Parse the validation rules per component if applicable
+        if node_validation_rules and isinstance(node_validation_rules, dict):
+            node_validation_rules = extract_component_validation_rules(
+                manifest_component=manifest_component,
+                validation_rules_dict=node_validation_rules,
+            )
+        return node_validation_rules
```

Comment:
The return type for this method is `list[str]` but the return type for `get_node_validation_rules` is list, which may get returned, and the return type for `extract_component_validation_rules` is list[Union[str, list]]`. Perpahs the returr type for this method should just be `list`